client/src/components/SignUp.js

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Redirect } from 'react-router';
+
+class SignUp extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            username: '',
+            password: '',
+            department: '',
+            registered: false
+        };
+    };
+
+
+    registerNewUser = e => {
+        var self = this;
```

Comment:
Keeping it consistent with ES6 syntax.

```suggestion
        let self = this;
```